src/webots/nodes/WbGroup.cpp

```diff
@@ -66,21 +66,33 @@ void WbGroup::downloadAssets() {
 void WbGroup::preFinalize() {
   WbBaseNode::preFinalize();
 
+  if (isWorldRoot()) {
+    emit worldLoadingStatusHasChanged(tr("Pre-finalizing nodes"));
+    mLoadProgress = 0;
+  }
+
   WbMFNode::Iterator it(*mChildren);
   while (it.hasNext()) {
+    mLoadProgress++;
     WbBaseNode *const n = static_cast<WbBaseNode *>(it.next());
     if (!mHasNoSolidAncestor) {
       WbGroup *group = dynamic_cast<WbGroup *>(n);
       if (group)
         group->mHasNoSolidAncestor = false;
     }
     n->preFinalize();
+    emit childFinalizationHasProgressed(mLoadProgress * 100 / (4 * childCount()));
```

Comment:
Did you consider adding a call to `QCoreApplication::processEvents(QEventLoop::AllEvents);` here to ensure the GUI is actually refreshed at this point?

Reply:
I don't believe it's necessary since
`childFinalizationHasProgressed` triggers `WbApplication::setWorldLoadingProgress` and in turn this signal 
[triggers](https://github.com/cyberbotics/webots/blob/2b385b0fb2a3bafc800eb70096199d03bcebefed/src/webots/gui/WbMainWindow.cpp#L145) `WbMainWindow::setWorldLoadingProgress` which processes the events:
https://github.com/cyberbotics/webots/blob/2b385b0fb2a3bafc800eb70096199d03bcebefed/src/webots/gui/WbMainWindow.cpp#L2380-L2385